pkg/reconcilermanager/controllers/permissions.go

```diff
@@ -25,11 +29,43 @@ func rolereference(name, kind string) rbacv1.RoleRef {
 	}
 }
 
-// subject returns and initialized Subject with kind, name and namespace.
-func subject(name, namespace, kind string) rbacv1.Subject {
+// newSubject returns and initialized Subject with kind, name and namespace.
+func newSubject(name, namespace, kind string) rbacv1.Subject {
 	return rbacv1.Subject{
 		Kind:      kind,
 		Name:      name,
 		Namespace: namespace,
 	}
 }
+
+// findSubjectIndex returns the index of the expected subject in the list, or -1.
+func findSubjectIndex(list []rbacv1.Subject, expected rbacv1.Subject) int {
+	for i, found := range list {
+		if found == expected {
+			return i
+		}
+	}
+	return -1
+}
+
+// addSubject adds a subject to the subject list.
+// Subjects are sorted, if modified.
+func addSubject(subjects []rbacv1.Subject, subject rbacv1.Subject) []rbacv1.Subject {
+	if i := findSubjectIndex(subjects, subject); i < 0 {
+		subjects = append(subjects, subject)
+		if len(subjects) > 0 {
```

Comment:
Did you mean to do `> 1`? I assume after an `append` the slice should always have nonzero length

Reply:
Good catch. There's no point sorting an array of size 1.

Reply:
fixed.